### PR TITLE
Preprocess support

### PR DESCRIPTION
**Pull request by omidontop**

### CODE REVIEW

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java [hunk 1]
```diff
@@ -141,24 +193,58 @@ private String formatParam(String param, String pathName) {
...
 		logger.debug("Temporary file placed at: {}", temp.getAbsolutePath());
 		fout.flush();
 		fout.close();
-		
+
 		return temp.getAbsolutePath();
 	}
+
+	private String retrieveSource(PublishJobOptions jobOptions) {
+		String jobName = jobOptions.getStorage().getPathconfigname();
```
Comment (takesson) [inline cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java]: Should not need to do this.

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java [hunk 2]
```diff
@@ -141,24 +193,58 @@ private String formatParam(String param, String pathName) {
...
+		jobName += jobOptions.getStorage().getPathdir();
+		jobName += "/" + jobOptions.getStorage().getPathnamebase();
+		CmsExportPrefix cmsPrefix = new CmsExportPrefix(jobOptions.getStorage().getPathversion());
+		String cloudId = jobOptions.getStorage().getPathcloudid();
+		String bucketName = jobOptions.getStorage().getParams().get("s3bucket");
+		CmsExportProviderAwsSingle exportProvider = new CmsExportProviderAwsSingle(cmsPrefix, cloudId, bucketName, region, credentials);
+		CmsImportJob downloadJob = new CmsImportJobSingle(null, jobName, "preprocess.zip");
```
Comment (takesson) [inline cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java]: Is it possible to use?
PublishExportJobFactory.getImportJobSingle(jobOptions.getStorage(), "preprocess.zip");

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/AwsStepfunctionPublishWorker.java [hunk 3]
```diff
@@ -168,6 +173,17 @@ public void run() {
 								updateStatusReport("Retrieving", new Date(), "Ticket: " + publishTicket.toString());
 								
 								String exportPath = exportCompletedJob(publishTicket, options);
+								// Delete the temporary directory if needed
```
Comment (takesson) [inline cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/AwsStepfunctionPublishWorker.java]: Can we move this code into PublishJobService? Perhaps into getCompletedJob(..) btw isComplete and the actual retrieval of the result.

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java [hunk 4]
```diff
@@ -60,65 +60,65 @@
...
             		bucketName = envBucket;
             	}
             	bind(bucketName).named("config:se.simonsoft.cms.publish.bucket").to(String.class);
             	cloudId = environment.getParamOptional("CLOUDID");
-            	
-            	region = Region.getRegion(Regions.fromName("eu-west-1")); // Currently hardcoded, might need different regions annotated per service. 
-        		// Might need to determine which region we are running in for EC2. See SDK 2.0, might have a Region Provider Chain.
-        		bind(region).to(Region.class);
+
+            	region = Region.getRegion(Regions.fromName("eu-west-1")); // Currently hardcoded, might need different regions annotated per service.
```
Comment (takesson) [inline cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java]: Oh, this is an old sin that I might be the author of. Can you try using DefaultAwsRegionProviderChain instead?

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java (hunk 1 above)
Comment (takesson) [reply cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java]: See the next comment.

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java (hunk 2 above)
Comment (takesson) [reply cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java]: In the POM, upgrade cms-publish-config to 0.5.0-SNAPSHOT. That will expose the factory.

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/AwsStepfunctionPublishWorker.java (hunk 3 above)
Comment (omidontop) [reply cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/AwsStepfunctionPublishWorker.java]: of course.

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java (hunk 2 above)
Comment (omidontop) [reply cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/PublishJobService.java]: Yep, done.

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java (hunk 4 above)
Comment (omidontop) [reply cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java]: No problem!

**Comment by omidontop** [comment]
@takesson I've now addressed all the review comments. Would you take a look?

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java [hunk 5]
```diff
@@ -107,10 +109,8 @@ protected void configure() {
             	}
             	bind(bucketName).named("config:se.simonsoft.cms.publish.bucket").to(String.class);
             	cloudId = environment.getParamOptional("CLOUDID");
-
-            	region = Region.getRegion(Regions.fromName("eu-west-1")); // Currently hardcoded, might need different regions annotated per service.
-            	// Might need to determine which region we are running in for EC2. See SDK 2.0,
-            	// might have a Region Provider Chain.
+            	region = RegionUtils.getRegion(new DefaultAwsRegionProviderChain().getRegion());
+            	logger.debug("Region name: {}", region.getName());
```
Comment (takesson) [inline cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java]: For safety (and backwards compat), add check for region == null with fallback to "eu-west-1" (and log an info statement).

**Comment by takesson** [comment]
Looks great! 

I added one comment on a fallback for Region. Will otherwise require an additional env var for on-prem (I think).

File: cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java (hunk 5 above)
Comment (omidontop) [reply cms-publish-worker-webapp/src/main/java/se/simonsoft/cms/publish/worker/startup/WorkerApplication.java]: Done!

**Pull request merged**